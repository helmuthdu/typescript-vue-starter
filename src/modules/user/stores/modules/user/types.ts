export enum UserActionTypes {
  IS_LOGGED_IN = 'USER/IS_LOGGED_IN',
  SET_USER = 'USER/SET_USER',
  DEL_USER = 'USER/DEL_USER',
  SIGN_UP = 'USER/SIGN_UP',
  SIGN_IN = 'USER/SIGN_IN',
  SIGN_OUT = 'USER/LOGOUT'
}
