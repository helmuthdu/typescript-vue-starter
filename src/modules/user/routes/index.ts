import { USER_ROUTE_PATH, USER_ROUTE_NAME } from '@/modules/user/routes/paths';
import { RouteRecordRaw } from 'vue-router';

export const routes: Array<RouteRecordRaw> = [
  {
    path: USER_ROUTE_PATH.ROOT,
    component: () => import('../layouts/empty.layout.vue'),
    children: [
      {
        path: USER_ROUTE_PATH.SIGN_IN,
        name: USER_ROUTE_NAME.SIGN_IN,
        component: () => import('./sign-in/sign-in.route.vue')
      }
    ]
  }
];
