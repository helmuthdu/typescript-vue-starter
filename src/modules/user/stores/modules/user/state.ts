export interface State {
  username: string;
  email: string;
  isLoggedIn: boolean;
  token: string;
}

export const state = (): State => ({
  username: '',
  email: '',
  isLoggedIn: false,
  token: ''
});
