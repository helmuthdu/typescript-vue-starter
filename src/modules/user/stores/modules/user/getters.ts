import { State as AppState } from '../../';
import { GetterTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export interface Getters<S, R> extends GetterTree<S, R> {
  [UserActionTypes.IS_LOGGED]: (state: S) => boolean;
}

export const getters: Getters<State, AppState> = {
  [UserActionTypes.IS_LOGGED](state) {
    return state.isLogged;
  }
};
