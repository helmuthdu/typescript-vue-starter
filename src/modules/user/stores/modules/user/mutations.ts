import { MutationTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export const mutations: MutationTree<State> = {
  [UserActionTypes.SET_USER](state: State, payload: State) {
    state.username = payload.username || '';
    state.email = payload.email || '';
    state.isLogged = payload.isLogged || false;
    state.token = payload.token || '';
  }
};
