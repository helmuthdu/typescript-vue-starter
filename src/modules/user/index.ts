import { routes } from './routes';
import * as stores from './stores';

export type State = Readonly<{
  [stores.user.name]: stores.user.State;
}>;

export { routes, stores };
