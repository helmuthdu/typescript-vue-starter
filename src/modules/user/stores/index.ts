import * as user from './modules/user';

export type State = Readonly<{
  [user.name]: user.State;
}>;

export { user };
