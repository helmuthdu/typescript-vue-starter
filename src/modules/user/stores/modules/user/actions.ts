import { State as AppState } from '../../';
import { userApi, AuthenticatePayload } from '../../../api/user.api';
import { ActionContext, ActionTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export interface Actions<S, R> extends ActionTree<S, R> {
  [UserActionTypes.LOGIN]: (context: ActionContext<S, R>, payload: any) => void;
  [UserActionTypes.LOGOUT]: (context: ActionContext<S, R>) => void;
}

export const actions: Actions<State, AppState> = {
  async [UserActionTypes.LOGIN]({ commit }, payload: AuthenticatePayload) {
    commit(UserActionTypes.SET_USER, {
      ...(await userApi.get(payload)).data,
      isLogged: true
    });
  },
  [UserActionTypes.LOGOUT]({ commit }) {
    commit(UserActionTypes.SET_USER, { username: '', email: '', isLogged: false });
  }
};
