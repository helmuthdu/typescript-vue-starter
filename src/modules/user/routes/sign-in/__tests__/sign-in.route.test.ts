import SignInRoute from '@/modules/auth/routes/sign-in/sign-in.route.vue';
import { UserActionTypes, State } from '../../../stores/modules/user';
import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';

const localVue = createLocalVue();

localVue.use(Vuex);
describe('Auth/Route -> SignIn', () => {
  let store: Store<State>;
  let wrapper: Wrapper<SignInRoute>;

  beforeEach(() => {
    store = new Vuex.Store({
      actions: {
        [UserActionTypes.LOGIN]: jest.fn()
      }
    });
    wrapper = shallowMount(SignInRoute, {
      store,
      localVue
    });
  });

  it('should renders with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
