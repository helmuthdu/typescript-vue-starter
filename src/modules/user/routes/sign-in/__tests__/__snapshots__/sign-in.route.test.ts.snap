// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth/Route -> SignIn should renders with props 1`] = `
<v-layout-stub tag="div" fill-height="" justify-center="" align-center="">
  <v-flex-stub tag="div" xs12="" sm5="" md4="" lg3="">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub primarytitle="" primary-title="">
        <v-layout-stub tag="div" column="">
          <v-flex-stub tag="div">
            <h2 class="headline">LOGIN</h2>
          </v-flex-stub>
          <v-flex-stub tag="div">
            <v-form-stub>
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Email" loaderheight="2" clearicon="$clear" type="text" required=""></v-text-field-stub>
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Password" loaderheight="2" clearicon="$clear" type="password" required=""></v-text-field-stub>
            </v-form-stub>
          </v-flex-stub>
        </v-layout-stub>
      </v-card-title-stub>
      <v-card-actions-stub>
        <v-container-stub tag="div">
          <v-layout-stub tag="div" column="" align-center="">
            <v-flex-stub tag="div">
              <v-btn-stub color="pink" dark="true" tag="button" activeclass="" type="button">Sign-in</v-btn-stub>
            </v-flex-stub>
          </v-layout-stub>
        </v-container-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-flex-stub>
</v-layout-stub>
`;
